core/ir/instr_api.h

```diff
@@ -722,6 +724,11 @@ DR_API
 int
 instr_get_opcode(instr_t *instr);
 
+DR_API
+/** Returns \p instr's set of categories (set of DR_INSTR_CATEGORY_ constants). */
```

Comment:
Continuing the discussion of this PR only setting categories on decode: what if I synthesize a new instruction and then ask about its categories?  The existing IR behavior would be to encode and figure things out from the encoding: that's how instr_length(), instr_zeroes_ymmh(), etc. all operate.  I would expect the same here.  At the very least this behavior of only being valid for unmodified-since-decode instructions would have to be clearly documented: but that route does not seem very desirable.